### PR TITLE
Prevent crashes from applying facility upgrades if the facility map isn't loaded

### LINKED ISSUE #775: Activating facility upgrades before facility maps are loaded crashes the game
Use case: making "debug strategy start" start with faction(s) met

Culprit: `class'Helpers'.static.SetOnAllActorsInLevelWithTag(none, //...`, called from `XGBase::UpdateFacilityUpgradeVisibilty`

### PR DESCRIPTION
**Pull request by Xymanek**
Fixes #775 

Better compare: https://github.com/X2CommunityCore/X2WOTCCommunityHighlander/compare/e5b0602...d2e6b8f

### CODE REVIEW

File: X2WOTCCommunityHighlander/Src/XComGame/Classes/XGBase.uc [hunk 1]
```diff
@@ -259,6 +259,11 @@ private function UpdateFacilityUpgradeVisibilty(int RoomIdx)
 	local string PreUpgradeName;
 	local XComGameState_HeadquartersRoom RoomState;
 
+	// Issue #775 (single line)
+	/// HL-Docs: ref:Bugfixes; issue:775
+	/// Prevent crashes from applying facility upgrades if the facility map isn't loaded
```
Comment (pledbrook) [inline X2WOTCCommunityHighlander/Src/XComGame/Classes/XGBase.uc:264]: Just a little thing: this sounds as if the *crashes* can apply facility upgrades. I understand what is meant, but it's awkwardly put (and this is part of the public documentation). Try:
> Applying/activating/enabling (choose which you prefer!) facility upgrades before the facility map is loaded no longer crashes the game.

File: X2WOTCCommunityHighlander/Src/XComGame/Classes/XGBase.uc [hunk 2]
```diff
@@ -259,6 +259,11 @@ private function UpdateFacilityUpgradeVisibilty(int RoomIdx)
...
+	if (m_arrLvlStreaming[RoomIdx] == none) return;
```
Comment (pledbrook) [inline X2WOTCCommunityHighlander/Src/XComGame/Classes/XGBase.uc:265]: I have no idea what the implications of this are, so don't feel qualified to say yes or no to it. What happens if a player tries to purchase the facility upgrade later? Or is it basically prepurchased?

Comment (Xymanek) [reply X2WOTCCommunityHighlander/Src/XComGame/Classes/XGBase.uc:265]: There is a very small window during strategy init where state is being processed but the visual aspect wasn't loaded yet. For changes to facilities in that window, the same logic as loading from a save will apply - the visual will init to the new state

**Review by robojumper — APPROVED** [review]

**Comment by pledbrook** [comment]
Merged with [this commit](https://github.com/X2CommunityCore/X2WOTCCommunityHighlander/commit/075ba86aa13284d00ba4f1e81e8f11a39df7b740) (I squashed the second and third commits).

**Pull request closed without merging**